### PR TITLE
Serverside paginated table

### PR DESCRIPTION
**Pull request by AndreaLandiArk**
Added Generic type table with support for 
Serverside Pagination, filtering and sorting.
Draggable reorderable columns

### CODE REVIEW

File: src/features/paginatedTable/movieTable.tsx [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+        placeholder={`Search... (${column.getFacetedUniqueValues().size})`}
+        className="w-36 border shadow rounded"
+        list={column.id + "list"}
+      />
+      <div className="h-1" />
+    </>
+  );
+}
+
+function DebouncedInput({
```
Comment (AndreaCuneo) [inline src/features/paginatedTable/movieTable.tsx:314]: this should be a common component reused across application.

File: README.md [hunk 2]
```diff
@@ -7,13 +7,17 @@ In this project, we've curated a collection of UI elements, layouts, and interac
 ## Packages
 
 In order to mantain this project with the most updated version of LTS's packages:
+
 Run `npm outdated` to see a table of packages with the current version, wanted version, and latest version.
+
 To update a specific package, you can use `npm update package_name` This will update the package to the 'wanted' version, which is the maximum version that satisfies the versioning range specified in package.json.
```
Comment (AndreaCuneo) [inline README.md:14]: I'd add suggestion to use [npx ](https://www.npmjs.com/package/npm-check-updates#installation) which is even better. I'd add it below, as the npm outdated is the 'official' one.

I'd add a strong suggestion to use npm-check-updates (ncu) via npx which is fantastic.

File: src/componentsCommon/chackraInputFilterWithClear/chackraInputFilterWithClear.tsx [hunk 3]
```diff
@@ -16,6 +16,10 @@ interface InputWithClearProps {
   title: string;
   propName: string;
 }
+export interface InputHeaderWithClearProps {
```
Comment (AndreaCuneo) [inline src/componentsCommon/chackraInputFilterWithClear/chackraInputFilterWithClear.tsx:19]: shouldn't this extend the whole InputProps to pass down the others option (ie. onFocus, etc.)?

File: src/componentsCommon/chackraInputFilterWithClear/chackraInputHeaderFilterWithClear.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import { Button, Input, InputGroup, InputRightElement } from "@chakra-ui/react";
+import { useState } from "react";
+import { TiTimes } from "react-icons/ti";
+import { InputHeaderWithClearProps } from "./chackraInputFilterWithClear";
+
+export const ChackraInputHeaderFilterWithClear: React.FC<
```
Comment (AndreaCuneo) [inline src/componentsCommon/chackraInputFilterWithClear/chackraInputHeaderFilterWithClear.tsx:6]: are you sure these components (which are nice) do play well with FinalForm? I think they should expose more parameters.

**Review by AndreaCuneo — APPROVED** [review]
aside few suggestion, the Table part looks grand and tested.

File: src/componentsCommon/chackraInputFilterWithClear/chackraInputHeaderFilterWithClear.tsx (hunk 4 above)
Comment (AndreaLandiArk) [reply src/componentsCommon/chackraInputFilterWithClear/chackraInputHeaderFilterWithClear.tsx:6]: this are for filters only, in final form config table vanilla inputs are being used iirc

Comment (AndreaLandiArk) [reply src/componentsCommon/chackraInputFilterWithClear/chackraInputHeaderFilterWithClear.tsx:6]: naming is bad i know

File: src/componentsCommon/PaginatedSortableTable/PaginatedSortableTable.tsx [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+            {isLoading || isFetching ? (
+              <Tr>
+                <Td colSpan={columns ? columns.length : 1}>
+                  <Center>
+                    <Spinner />
+                  </Center>
+                </Td>
+              </Tr>
+            ) : (
+              table.getRowModel().rows.map((row) => (
```
Comment (sean-sligo) [inline src/componentsCommon/PaginatedSortableTable/PaginatedSortableTable.tsx:214]: Could add a template with no date note when there is no data available

**Review by sean-sligo — APPROVED** [review]

**Review by sean-sligo — APPROVED** [review]

**Pull request merged**